### PR TITLE
[Openblas] Add Windows ARM64 support

### PR DESCRIPTION
**Pull request by nursik**
**Describe the pull request**

- #### What does your PR fix?
  Adds Windows ARM64 support

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Adds arm64-windows and arm64-windows-static. No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/openblas/portfile.cmake [hunk 1]
```diff
@@ -10,6 +10,7 @@ vcpkg_from_github(
         fix-redefinition-function.patch
         fix-uwp-build.patch
         install-tools.patch
+        arm64-win-generic.patch
```
Comment (Neumann-A) [inline ports/openblas/portfile.cmake:13]: Is this patch from upstream?

Comment (Neumann-A) [reply ports/openblas/portfile.cmake:13]: Maybe upstream it first?

Comment (Neumann-A) [reply ports/openblas/portfile.cmake:13]: > Upstreaming takes more time and OpenBLAS releases usually are made every 2 months.

doesn't need to reflect releases perfectly but adding architecture dependent patches should be validated by upstream unless specific to vcpkg for some reason.   

> This PR makes possible to unblock other OpenBLAS dependent VCPKG packages

#24622 also unblocks that by providing a Fortran compiler (flang-classic) for arm64 and using the blas implementation of Netlib instead.

**Comment by LilyWangLL** [comment]
@nursik Does this PR has any updates? `openblas:arm64-windows` and `openblas:arm64-uwp` build failed on CI pipeline.
The error message:
```
C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1432~1.313\bin\Hostx64\arm64\cl.exe    -ID:\buildtrees\openblas\src\50b9640a4e-eebf8a4915.clean -ID:\buildtrees\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   -DNO_LAPACK -DNO_LAPACKE -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.20\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\cblas_ctrmv.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c D:\buildtrees\openblas\arm64-windows-dbg\interface\CMakeFiles\cblas_ctrmv.c
D:\buildtrees\openblas\src\50b9640a4e-eebf8a4915.clean\common_arm64.h(95): warning C4013: '__rdtsc' undefined; assuming extern returning int
D:/buildtrees/openblas/src/50b9640a4e-eebf8a4915.clean/interface/ztrmv.c(248): error C2065: 'DTB_DEFAULT_ENTRIES': undeclared identifier
```

**Comment by LilyWangLL** [comment]
Ping @nursik for response. Is work still being done for this PR?

**Comment by LilyWangLL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Comment by nursik** [comment]
@LilyWangLL @Neumann-A The changes were approved in the upstream https://github.com/xianyi/OpenBLAS/pull/3642 The error was fixed and reflected in the patch.

**Comment by LilyWangLL** [comment]
> @LilyWangLL @Neumann-A The changes were approved in the upstream [xianyi/OpenBLAS#3642](https://github.com/xianyi/OpenBLAS/pull/3642) The error was fixed and reflected in the patch.

Now the fix patch has been merged in upstream, you can reopen this PR to continue fix support for ARM64.

**Comment by LilyWangLL** [comment]
@nursik Sorry, I don't know you can't reopen this, I opened this PR. You can continue to fix this.

**Comment by nursik** [comment]
I have checked the build process and unfortunately it only works with **VS 2022 Preview 3.** 

I would like to know if it is still appropriate to merge these changes and will be glad to hear any suggestions.

**Comment by nursik** [comment]
I have checked the build process and unfortunately it only works with **VS 2022 Preview 2**. 

I would like to know if it is still appropriate to merge these changes. Also, I will be glad to hear any suggestions.

**Comment by nursik** [comment]
The changes should work with Visual Studio 17.3.0 Preview 2. The build was done on Surface Pro with ARM chipset. Cross-compilation should work too.

**Comment by LilyWangLL** [comment]
I will test this again with Visual Studio 17.3.0 Preview 2.

**Comment by LilyWangLL** [comment]
@nursik I test this fix with Visual Studio 17.3.0 Preview 2, the build error is as below:
```
C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1433~1.315\bin\Hostx64\arm64\cl.exe    -IE:\Lily\openblas\buildtrees\openblas\src\50b9640a4e-9eb953cf83.clean -IE:\Lily\openblas\buildtrees\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   -DNO_LAPACK -DNO_LAPACKE -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.20\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\ztrmv.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c E:\Lily\openblas\buildtrees\openblas\arm64-windows-dbg\interface\CMakeFiles\ztrmv.c
E:\Lily\openblas\buildtrees\openblas\src\50b9640a4e-9eb953cf83.clean\common_arm64.h(101): warning C4244: 'function': conversion from 'const int64_t' to 'int', possible loss of data
E:/Lily/openblas/buildtrees/openblas/src/50b9640a4e-9eb953cf83.clean/interface/ztrmv.c(248): error C2065: 'DTB_DEFAULT_ENTRIES': undeclared identifier
```

**Comment by nursik** [comment]
@LilyWangLL Could you please confirm build failure using Preview 3?

**Comment by LilyWangLL** [comment]
@nursik The error is same with Visual Studio 17.3.0 Preview 2.
```
[33/1587] C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1433~1.316\bin\Hostx64\arm64\cl.exe    -IE:\Lily\openblas\buildtrees\openblas\src\50b9640a4e-9eb953cf83.clean -IE:\Lily\openblas\buildtrees\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   -DNO_LAPACK -DNO_LAPACKE -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.20\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\ztrmv.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c E:\Lily\openblas\buildtrees\openblas\arm64-windows-dbg\interface\CMakeFiles\ztrmv.c
FAILED: interface/CMakeFiles/interface.dir/CMakeFiles/ztrmv.c.obj 
C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1433~1.316\bin\Hostx64\arm64\cl.exe    -IE:\Lily\openblas\buildtrees\openblas\src\50b9640a4e-9eb953cf83.clean -IE:\Lily\openblas\buildtrees\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   -DNO_LAPACK -DNO_LAPACKE -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.20\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\ztrmv.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c E:\Lily\openblas\buildtrees\openblas\arm64-windows-dbg\interface\CMakeFiles\ztrmv.c
E:\Lily\openblas\buildtrees\openblas\src\50b9640a4e-9eb953cf83.clean\common_arm64.h(101): warning C4244: 'function': conversion from 'const int64_t' to 'int', possible loss of data
E:/Lily/openblas/buildtrees/openblas/src/50b9640a4e-9eb953cf83.clean/interface/ztrmv.c(248): error C2065: 'DTB_DEFAULT_ENTRIES': undeclared identifier
```

**Comment by nursik** [comment]
@LilyWangLL There are changes to fix the issue of cross-compilation. These changes also have been approved in the [upstream](https://github.com/xianyi/OpenBLAS/pull/3709).

**Comment by LilyWangLL** [comment]
> @LilyWangLL There are changes to fix the issue of cross-compilation. These changes also have been approved in the [upstream](https://github.com/xianyi/OpenBLAS/pull/3709).

You can sync up this upstream change to this PR, I can test again.

**Comment by nursik** [comment]
@LilyWangLL I am sorry, what do you mean by syncing the changes?

**Comment by LilyWangLL** [comment]
Sorry I missed your new commit, I will test the new change again.

**Comment by nursik** [comment]
@LilyWangLL Thanks for the review and also I also removed `arm-uwp` support.

**Comment by nursik** [comment]
@LilyWangLL I returned arm-uwp as previously it was in the support clause. Also fixed SHA version so x86-windows does not fail anymore. If there is anything to be done in order to approve and merge the changes, I would like to hear your suggestions.

**Comment by nursik** [comment]
@LilyWangLL Removed arm-uwp again and added dependent packages (simbody, geogram, faiss, dlib) in baseline.txt, because they fail during CI/CD builds.

**Comment by LilyWangLL** [comment]
> @LilyWangLL Could you please review final changes?

Sorry, I missed this message, I will test and reply today.

**Review by LilyWangLL — APPROVED** [review]

**Comment by LilyWangLL** [comment]
I test `openblas:arm64-windows` pass locally, but I can't confirm add these ports `arm64-windows` failed which depend on `openblas` in **ci.baseline.txt** is accept. I tagged this PR `reviewed`, waiting for other members confirm and merge this PR.

**Comment by nursik** [comment]
Other packages fail because their vcpkg.json files did not have `!(arm & windows)` expression and because they were dependent on OpenBLAS they were not failing in CI/CD previously. Thus, I added them into ci.baseline.txt rather than creating multiple PRs with adding above expression into vcpkg.json files.

**Comment by nursik** [comment]
@JavierMatosD Created a [PR](https://github.com/microsoft/vcpkg/pull/26396) for OpenBLAS upgrade. Closing this PR as it is as a duplicate (All work is reflected in upstream repository).

**Pull request closed without merging**